### ISSUE TITLE
Experiment on the XD-Violence dataset

### ISSUE THREAD

**Issue by DungVo1507** [tianyu0207/RTFM#43]
Hi @tianyu0207,
I loaded the i3d RGB feature of the XD-Violence dataset, but it uses 5 crops instead of 10 crops like you said.
So do I need to fix 10 to 5 in the below two command lines in model.py?
`normal_features = features[0:self.batch_size*10]`
`abnormal_features = features[self.batch_size*10:]`
And when you experiment do you use 5 crops or 10 crops?

Thank you in advance!

**Comment by tianyu0207** [comment]
> Hi @tianyu0207,
> I loaded the i3d RGB feature of the XD-Violence dataset, but it uses 5 crops instead of 10 crops like you said.
> So do I need to fix 10 to 5 in the below two command lines in model.py?
> `normal_features = features[0:self.batch_size*10]`
> `abnormal_features = features[self.batch_size*10:]`
> And when you experiment do you use 5 crops or 10 crops?
> 
> Thank you in advance!

Hi For xd-violence, feel free to use the 5 crops and change that to normal_features = features[0:self.batch_size*5]. You should achieve similar results in the paper.

**Comment by DungVo1507** [comment]
Thank you @tianyu0207,
I have implemented it on the XD-Violence dataset but the result in your paper is the AP measure, not the AUC.
Can you tell me what I need to modify in test_10crop.py to get the AP result?
Here is the original code:
` gt = np.load(args.gt)

        pred = list(pred.cpu().detach().numpy())
        pred = np.repeat(np.array(pred), 16)
        #print(pred.shape)
        fpr, tpr, threshold = roc_curve(list(gt), pred)
        np.save('fpr.npy', fpr)
        np.save('tpr.npy', tpr)
        rec_auc = auc(fpr, tpr)
        print('auc : ' + str(rec_auc))

        precision, recall, th = precision_recall_curve(list(gt), pred)
        pr_auc = auc(recall, precision)
        np.save('precision.npy', precision)
        np.save('recall.npy', recall)
        viz.plot_lines('pr_auc', pr_auc)
        viz.plot_lines('auc', rec_auc)
        viz.lines('scores', pred)
        viz.lines('roc', tpr, fpr)
        return rec_auc`

**Comment by DungVo1507** [comment]
Thank you @tianyu0207,
I implemented it on the XD-Violence dataset but the results in your paper are the AP measure, not the AUC.
I have 2 questions hope you can help with:

- Could you please send me the ground truth of XD-Violence?
- Can you tell me what I need to modify in test_10crop.py to get AP results?

Grateful to you!
